pkg/controller/upgradeconfig/upgradeconfig_controller.go

```diff
@@ -272,10 +272,12 @@ func (r *ReconcileUpgradeConfig) Reconcile(ctx context.Context, request reconcil
 		return r.upgradeCluster(upgrader, instance, reqLogger)
 	case upgradev1alpha1.UpgradePhaseUpgraded:
 		reqLogger.Info("Cluster is already upgraded")
-		return reconcile.Result{}, nil
+		err = reportUpgradeMetrics(metricsClient, instance.Name, instance.Spec.Desired.Version, history.StartTime.Time, history.CompleteTime.Time)
+		return reconcile.Result{}, err
 	case upgradev1alpha1.UpgradePhaseFailed:
 		reqLogger.Info("Cluster has failed to upgrade")
-		return reconcile.Result{}, nil
+		err = reportUpgradeMetrics(metricsClient, instance.Name, instance.Spec.Desired.Version, history.StartTime.Time, history.CompleteTime.Time)
```

Comment:
May be worth doublechecking this, but in the event of an upgradefailure I don't believe `CompleteTime` is set (I see only one instance that writes to in the code, and that's when the phase is `Upgraded`), and I'm not sure how that might throw off the `reportUpgradeMetrics` function.